### DISCUSSION TITLE
I'm not understanding Git variables

### DISCUSSION THREAD

**Discussion by simsrw73** [JanDeDobbeleer/oh-my-posh#3259] [Troubleshoot]
I'm trying to mimic the [Powerlevel10k git status](https://github.com/romkatv/powerlevel10k/blob/master/README.md#what-do-different-symbols-in-git-status-mean) for my [Catppuccin Themes](https://github.com/simsrw73/poshcat.omp). Specifically, the last lines of that table: *42 ~42 +42 !42 ?42). But I'm not understanding the variables supplied by OMP. I'm not seeing any change to the the *.Untracked & *.Unmerged variables. They are always 0. I'm not a git guru. Barely passible, so maybe I'm misunderstanding? I do have `$env:POSH_GIT_ENABLED = $true` and the git segment of all the themes I've tried seems to work. What am I doing wrong?

**Comment by JanDeDobbeleer** [comment]
`$env:POSH_GIT_ENABLED = $true` is only relevant when using posh-git, not really needed when using plain git. As for the others, they will be 0 when there aren't any untracked or unmerged files. However, we do not sum all staged and unstaged files like they do. I could create that specifically as doing that in a template will be rather cumbersome. Allow me to create a function in the git segment that outputs the same, ok?

**Reply by simsrw73** [reply]
I think I'm a little closer to where I need to be with the example below. I still don't understand the `.Untracked` and `.Unmerged` and `.StashCount` vars.

For untracked changes, I tested by cloning any repo and adding a file (`touch foo`) and checking the values of the various variables. *.Untracked never changes value. However, it seems like I can use `{{ .Working.Added }}` to get the same info that p10k displays for untracked. That is the same info that I expected `.Working.Untracked` to display, but maybe I'm misunderstanding?

Unstaged changes in p10k seem to be equivalent to `{{ add .Working.Deleted .Working.Modified }}` in OMP.

For unmerged, I guess I'm not sure how to create a test case to check this.

I'm also not get any value change on `.StashCount`. Making any change in a repo and running `git stash` triggers the indicator for p10k but not OMP, given the example below.

Haven't tested yet on `.WorktreeCount`. I've never used that feature and I don't think p10k tracks it. May remove it, but I want to understand it first.

```
{
    "type": "git",
    "properties": {
        "fetch_status": true,
        "fetch_stash_count": true,
        "fetch_working_count": true
    },
    "style": "plain",
    "background": "p:Mantle",
    "foreground": "p:Green",
    "templates": [
        "<p:Overlay2>\uE0B1</> {{ .HEAD }} ",
        "{{ if .BranchStatus }}<p:Green>{{ .BranchStatus }}</> {{ end }}",
        "{{ if .Staging.Changed }}<p:Yellow>+{{ add .Staging.Deleted .Staging.Added .Staging.Modified }}</> {{ end }}",
        "{{ if add .Working.Deleted .Working.Modified }}<p:Yellow>!{{ add .Working.Deleted .Working.Modified }}</> {{ end }}",
        "{{ if .Working.Added }}<p:Sky>?{{ .Working.Added }}</>{{ end }} ",
        "{{ if gt .StashCount 0 }}<p:Pink>*\uF692</> {{ .StashCount }} {{ end }}  *{{ .StashCount }}",
        "{{ if gt .WorktreeCount 0 }}<p:Pink>\uF1BB</> {{ .WorktreeCount }} {{ end }}"
    ]
},
```

**Reply by JanDeDobbeleer** [reply]
@simsrw73 StashCount requires a switch, but I have a commit ready to remove the need for that. See `fetch_stash_count` [here](https://ohmyposh.dev/docs/segments/git#fetching-information), same for worktrees btw. I don't see why untracked doesn't work, works here:

![image](https://user-images.githubusercontent.com/2492783/209839496-adeafce9-2ade-4609-8602-2c7fdd01a8b1.png)
